server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -710,9 +746,82 @@ public Exception getInboundException() {
         assertThat(channel.getRestResponse().getHeaders().get("Allow"), hasItem(equalTo(GET.toString())));
     }
 
+    /**
+     * Check that when dispatching a request, if an IllegalArgumentException is thrown, then a trace span is started
+     * and the exception is captured in the span.
+     */
+    public void testDispatchUnsupportedHttpMethodTracesException() {
+        final RestRequest request = RestRequest.request(parserConfig(), new HttpRequest() {
```

Comment:
Can't we override `FakeHttpRequest` here? Seems we only want the `method()` override right?